actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java

```diff
@@ -110,4 +119,122 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
+        batchList.add(map);
+        openBatchMap.put(courseId, batchList);
+      }
+    } else {
+      if (privateBatchMap.containsKey(courseId)) {
+        privateBatchMap.get(courseId).add(map);
+      } else {
+        List<Map<String, Object>> batchList = new ArrayList<>();
+        batchList.add(map);
+        privateBatchMap.put(courseId, batchList);
+      }
+    }
+  }
+
+  private void contentMapUpdate(boolean increment) {
+    for (Map.Entry<String, List<Map<String, Object>>> openBatchList : openBatchMap.entrySet()) {
+      String contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.OPEN);
+      updateContentMapCount(
+          increment, contentName, openBatchList.getKey(), openBatchList.getValue().size());
+    }
+    for (Map.Entry<String, List<Map<String, Object>>> privateBatchList :
+        privateBatchMap.entrySet()) {
+      String contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.INVITE_ONLY);
+      updateContentMapCount(
+          increment, contentName, privateBatchList.getKey(), privateBatchList.getValue().size());
+    }
+    updateEkstepAndDb();
+  }
+
+  private void updateContentMapCount(
+      boolean increment, String contentName, String courseId, int size) {
+    int val = (int) contentMap.get(courseId).getOrDefault(contentName, 0);
+    if (increment) {
+      val += size;
+    } else {
+      val -= size;
+      if (val < 0) {
+        val = 0;
+      }
+    }
+    contentMap.get(courseId).put(contentName, val);
+  }
+
+  private void getContentAndStore(Map<String, List<Map<String, Object>>> batchMap) {
+    Map<String, String> ekstepHeader = CourseBatchSchedulerUtil.headerMap;
+    for (Map.Entry<String, List<Map<String, Object>>> batchList : batchMap.entrySet()) {
+      String courseId = batchList.getKey();
+      if (!contentMap.containsKey(courseId)) {
+        Map<String, Object> ekStepContent =
+            CourseEnrollmentActor.getCourseObjectFromEkStep(courseId, ekstepHeader);
+        if (null != ekStepContent && ekStepContent.size() > 0) {
+          contentMap.put(courseId, ekStepContent);
+        }
+      }
+    }
+  }
+
+  private void updateEkstepAndDb() {
+    boolean response;
+    for (Map.Entry<String, Map<String, Object>> ekStepUpdateMap : contentMap.entrySet()) {
+      String courseId = ekStepUpdateMap.getKey();
+      String contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.OPEN);
+      response =
+          CourseBatchSchedulerUtil.updateEkstepContent(
+              courseId, contentName, (int) ekStepUpdateMap.getValue().get(contentName));
+      if (response) {
+        openBatchMap
+            .get(courseId)
+            .forEach(
+                map -> {
+                  if (CourseBatchSchedulerUtil.updateDataIntoES(map)) {
+                    CourseBatchSchedulerUtil.updateDataIntoCassandra(map);
+                  }
+                });
+        contentName = CourseBatchSchedulerUtil.getContentName(JsonKey.INVITE_ONLY);
```

Comment:
again similar repetion - veriability is invite_only vs open -rest of the code is pretty much the same.